wrappers/services.go

```diff
@@ -471,13 +471,48 @@ func tryFileUpdate(path string, desiredContent []byte) (old *osutil.MemoryFileSt
 	}
 }
 
+type ServiceQuotaMap map[*snap.AppInfo]*quota.Group
+
+// MakeServiceQuotaMap builds a map of services and their quota groups. The goal is
+// to make sure that if any quota group applies to a snap service, then we can easily
+// look it up in this based on it's snap.AppInfo. We take the snap instance name, the services
+// want to include in the map (should only be services of the snap name provided) and the quota
+// group of the snap.
+func MakeServiceQuotaMap(snapName string, svcs []*snap.AppInfo, grp *quota.Group) ServiceQuotaMap {
+	if len(svcs) == 0 {
+		return nil
+	}
+
+	svcQuotaMap := make(ServiceQuotaMap, len(svcs))
+	for _, svc := range svcs {
+		// set nil grps for all services if parent is nil
+		if grp == nil {
+			svcQuotaMap[svc] = nil
+			continue
+		}
+
+		// always default to the snap quota group if the service is not
+		// in a separate one
+		svcGrp := grp.FindQuotaGroupForServiceInSubGroups(fmt.Sprintf("%s.%s", snapName, svc.Name))
+		if svcGrp == nil {
+			svcGrp = grp
```

Comment:
it's a bit unclear to me whether this should happen here, or a a level up, it relates also about the question about the field in QuotaGroup field in SnapSeviceOptions

Reply:
I think it makes sense to have it here, it allows for a bit of cleaner code in wrappers.